modules/android/tests/test_config_props.lua

```diff
@@ -0,0 +1,73 @@
+	local p = premake
+	local suite = test.declare("android_config_props")
+	local vc2010 = p.vstudio.vc2010
+	local project = p.project
+
+
+--
+-- Setup
+--
+
+	local wks, prj
+
+	function suite.setup()
+		p.action.set("vs2015")
+		wks, prj = test.createWorkspace()
+	end
+
+	local function prepare()
+		system "android"
+		local cfg = test.getconfig(prj, "Debug")
+		vc2010.configurationProperties(cfg)
+	end
+
+
+--
+-- Check the structure with the default project values.
+--
+
+	function suite.structureIsCorrect_onDefaultValues()
+		prepare()
+		test.capture [[
+<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Android'" Label="Configuration">
```

Comment:
This doesn't seem right to me, the value for `$(Platform)` was always `x86`, `x64`, `ARM` or `ARM64` for me, why is it `Android` now?

Reply:
Not my code! The tests run ;)

Reply:
How can Android use the arch as the platform, and not be confused with windows configurations (that are also named for the arch)?

Reply:
There's something wrong with the test cases in general.  We already had test cases for rtti and there a bug fixes here specifically for rtti.  Somehow the tests are working differently in isolation and it might be why the platform is weird here too.

Reply:
Yeah, this looks like the vs-android configuration. Microsoft uses the platform to indicate the architecture, not the system - they use `ApplicationType` to indicate the system. I would say that the action isn't being set correctly for the tests, it does seem to set `p.action.set("vs2015")` in the `suite.setup() ` which mightn't be the correct way of doing this in the unit tests. Might be worth giving "vs2015" a search in the tests folder and seeing how it's done elsewhere. :+1: